CatalogService/Controllers/CatalogServiceController.cs

```diff
@@ -0,0 +1,155 @@
+using CatalogService.Entities;
+using CatalogService.Repository;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace CatalogService.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]/v1")]
+    public class CatalogServiceController : ControllerBase
+    {
+        private readonly ILogger<CatalogServiceController> _logger;
+        private ICategoryRepository _repository;
+
+        public CatalogServiceController(ILogger<CatalogServiceController> logger, ICategoryRepository repository)
+        {
+            _logger = logger;
+            _repository = repository;
+        }
+
+        [Route("categories")]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public ActionResult<List<Category>> GetCategories()
+        {
+            try
+            {
+                var categories = _repository.GetCategories();
+                var json = JsonConvert.SerializeObject(categories);
+                return Ok(json);
+            }
+            catch {
```

Comment:
1. Do we expect that all exceptions are caused by the invalid user's input? 
Note: the same logic can be applied to other catch blocks in this class as well.
2. Formatting can be fixed